plugins/become/machinectl.py

```diff
@@ -66,15 +66,37 @@
             ini:
               - section: machinectl_become_plugin
                 key: password
+    notes:
+      - This plugin requires a polkit rule which will alter the behaviour of machinectl to ask directly for the user
+        credentials if the user is allowed to perform the action
+      - >
+        An example for such a polkit role would be C(
+        polkit.addRule(function(action, subject) {
+          if(action.id == "org.freedesktop.machine1.host-shell" && subject.isInGroup("wheel")) {
+              return polkit.Result.AUTH_SELF_KEEP;
+          }
+        });)
 '''
 
+from re import compile as re_compile
+
 from ansible.plugins.become import BecomeBase
 
 
 class BecomeModule(BecomeBase):
 
     name = 'community.general.machinectl'
 
+    prompt = 'Password: '
+    fail = ('==== AUTHENTICATION FAILED ====',)
+    success = ('==== AUTHENTICATION COMPLETE ====',)
+
+    @staticmethod
+    def remove_ansi_codes(line):
+        # taken from https://stackoverflow.com/a/38662876/9531111
+        ansi_escape = re_compile(rb'(?:\x1B[@-_]|[\x80-\x9F])[0-?]*[ -/]*[@-~]')
```

Comment:
You cannot combine `r` and `b` prefixes in Python 2. Use the following:
```suggestion
        ansi_escape = re_compile(to_bytes(r'(?:\x1B[@-_]|[\x80-\x9F])[0-?]*[ -/]*[@-~]'))
```
For this you need `from ansible.module_utils.common.text.converters import to_bytes`.